// design-system/index.js
/*
    genre.guide - Design system information JavaScript file
    Copyright (C) 2020 Navith

    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU Affero General Public License as published by
    the Free Software Foundation, either version 3 of the License, or
    (at your option) any later version.

    This program is distributed in the hope that it will be useful,
    but WITHOUT ANY WARRANTY; without even the implied warranty of
    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
    GNU Affero General Public License for more details.

    You should have received a copy of the GNU Affero General Public License
    along with this program. If not, see <https://www.gnu.org/licenses/>.
*/

import {
    quartIn, quartInOut, quartOut, sineIn, sineInOut, sineOut,
} from "svelte/easing";
import { fade } from "svelte/transition";

export const easingFunctions = {
    smooth: {
        in: sineIn,
        out: sineOut,
        inOut: sineInOut,
    },
    smoother: {
        in: quartIn,
        out: quartOut,
        inOut: quartInOut,
    },
};

export const transitionDurations = {
    short: 200,
    medium: 300,
};

export const transitionFunctions = {
    fade,
};
